src/thor/route_matcher.cc

```diff
@@ -184,12 +201,14 @@ bool expand_from_node(const std::shared_ptr<DynamicCost>* mode_costing,
         prev_edge_label = {kInvalidLabel, edge_id, de, {}, 0, 0, mode, 0};
 
         // Continue walking shape to find the end edge...
-        if (expand_from_node(mode_costing, mode, reader, shape, distances, use_timestamps, index,
-                             end_node_tile, de->endnode(), end_nodes, prev_edge_label, elapsed,
-                             path_infos, false, end_node, followed_edges)) {
+        if (expand_from_node(mode_costing, mode, reader, shape, distances, origin_epoch,
+                             use_timestamps, index, end_node_tile, de->endnode(), end_nodes,
+                             prev_edge_label, elapsed, path_infos, false, end_node, followed_edges)) {
           return true;
         } else {
-          // Match failed along this edge, pop the last entry off path_infos and try the next edge
+          // Match failed along this edge, pop the last entry off path_infos as well as what it
+          // contributed to the elapsed cost/time and try to keep going on the next edge
+          elapsed -= cost;
```

Comment:
previously when we went down a path that didnt match we weren't removing the impact it had on the elapsed time. this should take care of that and was the only reason i broke out the cost into its own object above. another option would ahve been to look at the previous path info if any and do the subtracting from there